src/handlers/cell.ts

```diff
@@ -21,31 +21,42 @@ const LINE_HIGHLIGHT_CLASS = 'jp-breakpoint-line-highlight';
 
 export class CellManager implements IDisposable {
   constructor(options: CellManager.IOptions) {
-    this._debuggerModel = options.debuggerModel;
     this._debuggerService = options.debuggerService;
-    this.breakpointsModel = options.breakpointsModel;
+    this.onModelChanged();
+    this._debuggerService.modelChanged.connect(() => this.onModelChanged());
     this.activeCell = options.activeCell;
     this.onActiveCellChanged();
+  }
 
-    this._debuggerModel.variablesModel.changed.connect(() => {
-      this.cleanupHighlight();
-      const firstFrame = this._debuggerModel.callstackModel.frames[0];
-      if (!firstFrame) {
-        return;
-      }
-      this.showCurrentLine(firstFrame.line);
-    });
+  isDisposed: boolean;
```

Comment:
Should we try to follow https://github.com/jupyterlab/debugger/issues/139 as of now and add a docstring? or later?

Reply:
let's do it later for the whole file in a "refactoring" pass.

Reply:
:+1: